src/net_processing.cpp

```diff
@@ -731,6 +735,37 @@ std::chrono::microseconds CalculateTxGetDataTime(const uint256& txid, std::chron
     return process_time;
 }
 
+static void RetryProcessTx(CConnman& connman, const uint256& txid, const std::chrono::microseconds current_time) EXCLUSIVE_LOCKS_REQUIRED(cs_main)
+{
+    CNodeState::TxDownloadState* best_d = nullptr;
+    std::chrono::microseconds best;
+
+    for (auto& el : mapNodeState) {
```

Comment:
Did you consider to filter by `nPreferredDownload` to favor outbound peers and lower DoS risk? Only querying outbound peers should be good enough to achieve goal of finding NOTFOUND transactions but even if we don't success due to bad-connectivity of our outbounds, worst-case scenario we hit the 1-min window (and a transaction not being announced by our outbounds is less likely to be a honest announcement?)

Reply:
I think it'd be even better to change the 'best' definition to prefer outbound peers where we can, but fall back to inbound peers if there are no outbound peers that have announced the tx.

Reply:
We already prefer outbound peers when we bump the process time (in `CalculateTxGetDataTime()`), and this chooses the lowest process time which will reflect that preference. I don't think it makes sense to complicate this further unless someone wants to do some real world testing on how well/badly the process time preferencing works in practice when the first thing we try results in a notfound.